### PR TITLE
docs: update DI in action to recommend new way of defining providers

### PR DESCRIPTION
**Pull request by sreevani-ship-it**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Does not describe the new way of defining providers
Issue Number: N/A

## What is the new behavior?
added doc that describes the new way of defining providers

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: aio/content/guide/dependency-injection-in-action.md [hunk 1]
```diff
@@ -142,14 +130,6 @@ Notice the `@Injectable()`decorator on the `UserContextService` class.
 
 That decorator makes it possible for Angular to identify the types of its two dependencies, `LoggerService` and `UserService`.
```
Comment (alxhub) [inline aio/content/guide/dependency-injection-in-action.md]: > That decorator indicates that the Angular DI system will be used to create one or more instances of `UserContextService`.

File: aio/content/guide/dependency-injection-in-action.md [hunk 2]
```diff
@@ -142,14 +130,6 @@ Notice the `@Injectable()`decorator on the `UserContextService` class.
...
-Technically, the `@Injectable()`decorator is only required for a service class that has _its own dependencies_.
-The `LoggerService` doesn't depend on anything. The logger would work if you omitted `@Injectable()`
-and the generated code would be slightly smaller.
-
-But the service would break the moment you gave it a dependency and you'd have to go back
-and add `@Injectable()` to fix it. Add `@Injectable()` from the start for the sake
-of consistency and to avoid future pain.
-
 
 <div class="alert is-helpful">
```
Comment (alxhub) [inline aio/content/guide/dependency-injection-in-action.md]: Delete this `alert-is-helpful`, it's factually incorrect. The `l-sub-section` following it (about why `AppComponent` doesn't need `Injectable`) can also go away.

**Review by alxhub — CHANGES_REQUESTED** [review]

**Review by alxhub — APPROVED** [review]

**Review by jenniferfell — APPROVED** [review]

**Comment by jenniferfell** [comment]
@IgorMinar Would you please approve this? It's asking for someone from io-guide-and-tutorial. Alex R and I have approved. Just need to pass pull-approve test now. Thank you!

**Comment by jenniferfell** [comment]
@IgorMinar Can you also provide an approval. We need an approval from you or someone else in the guide-and-tutorial group. This PR doesn't touch tutorial files, just ngmodule and DI files, but pull-approve is asking for the guide-and-tutorial group. AlexR has approved. Thanks!

**Review by IgorMinar — APPROVED** [review]

**Pull request closed without merging**